Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigation.m

```diff
@@ -149,34 +150,36 @@ - (void)updateSelection:(UIButton *)sender {
     }
 }
 
-- (BPKHorizontalNavigationItem *)createHorizontalNavigationItemWithDefinition:
-    (BPKHorizontalNavigationOption *)definition {
-    BPKHorizontalNavigationItem *newItem = [[BPKHorizontalNavigationItem alloc] initWithName:definition.name
-                                                                                    iconName:definition.iconName];
+- (UIControl<BPKHorizontalNavigationItem> *)createHorizontalNavigationItemWithDefinition:
+    (id<BPKHorizontalNavigationOptionType>)definition {
+    UIControl<BPKHorizontalNavigationItem> *newItem = [definition makeItem];
     newItem.tag = definition.tag;
+
     return newItem;
 }
 
-- (void)forEachNavigationItem:(void (^)(BPKHorizontalNavigationItem *))callback {
+- (void)forEachNavigationItem:(void (^)(UIControl<BPKHorizontalNavigationItem> *))callback {
     for (UIView *subView in self.stackView.arrangedSubviews) {
-        NSAssert([subView isKindOfClass:[BPKHorizontalNavigationItem class]],
-                 @"HorizontalNav subview is not of type BPKHorizontalNavigationItem as expected.");
-        if (![subView isKindOfClass:[BPKHorizontalNavigationItem class]]) {
+        NSAssert([subView isKindOfClass:[UIControl class]] &&
```

Comment:
Do we still need to check for this inheritance and protocol? can we have some scenario in this current implementation where you can have one of the subviews not being a `UIControl` or conforming the `BPKHorizontalNavigationItem` protocol?

Reply:
It's because we are fishing them out of the stack views arranged subviews here. In theory there should only be things of this type in the stack view but it doesn't hurt to have the check just in case?

Reply:
the thing is that AFAIK the `stackView` is not exposed in the `BPKHorizontalNavigation` so I think that there's not a direct way to jump that rule and do what you've mentioned right?

Reply:
Future us can make a mistake